core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -189,6 +189,21 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
       );
   }
 
+  rearrangeLanguageName(language: string): string {
+    // Check if the language contains the expected format 'abc (something)'.
+    if (language.includes('(') && language.includes(')')) {
+      // Split the language into parts using '(' and ')' as separators
```

Comment:
Instead of doing all this could you just construct a regex for the desired format and check whether the string matches it? You can use capture groups to collect the relevant parts. I think this would lead to cleaner code and less string manipulation.

Reply:
@seanlip I am using `RegExp(/([\p{L}]+)\s*\(([\p{L}]+)\)/, 'u')` but getting linter error.

Reply:
Please provide the error you see, it's difficult to help without understanding the error you get.

Reply:
https://pastebin.com/Tzyvu9qK

Reply:
@Aakash-Jakhmola The error message says `bad escape \p at position 2` but I'm not sure what's causing that. It seems to work fine when I try it here ... https://esprima.org/demo/validate.html

https://bugs.eclipse.org/bugs/show_bug.cgi?id=492025 suggests that correctness is determine by the "runtime's RegExp impl" but I'm not sure what that's referring to exactly.

I wonder if the issue is that whatever impl is being used doesn't recognize the \p{L}. Could you try replacing that with "accept any character" and see if that helps? Also btw the second part of the regex should just be [A-Za-z ], right?